### PR TITLE
Do a canary ping while connecting, pass info to the unavailable notification

### PR DESCRIPTION
**Pull request by strseb**
## Description
Fixes: Server unavailable message is misleading if blocked by firewall

-> In parallel with the handshake, we attempt to ping the server.
-> If the ping succeeds and the handshake fails, the error message is changed to reflect that this might be a firewall issue. 

Screenshot 2022-08-29 160220

## Reference

[ Jira VPN-1748 ](https://mozilla-hub.atlassian.net/browse/VPN-1748)

## Checklist
    
- [x] My code follows the style guidelines for this project
- [x] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [x] I have performed a self review of my own code
- [x] I have commented my code PARTICULARLY in hard to understand areas
- [ ] I have added thorough tests where needed

### CODE REVIEW

File: nebula/ui/components/VPNServerUnavailablePopup.qml [hunk 1]
```diff
@@ -10,12 +10,14 @@ import Mozilla.VPN 1.0
 
 VPNSimplePopup {
     id: root
+    property bool recivedPing: false;
```
Comment (bakulf) [inline nebula/ui/components/VPNServerUnavailablePopup.qml:13]: received?

File: src/controller.cpp [hunk 2]
```diff
@@ -122,6 +123,12 @@ void Controller::initialize() {
   connect(this, &Controller::stateChanged, this,
           &Controller::maybeEnableDisconnectInConfirming);
 
+  connect(&m_ping_canary, &PingHelper::pingSentAndReceived, [this]() {
+    m_ping_canary.stop();
+    m_ping_recieved = true;
```
Comment (bakulf) [inline src/controller.cpp:128]: received?

File: src/controller.cpp [hunk 3]
```diff
@@ -13,6 +13,7 @@
 #include "models/feature.h"
 #include "models/server.h"
 #include "mozillavpn.h"
+#include "pinghelper.h"
```
Comment (bakulf) [inline src/controller.cpp:16]: not needed

**Review by bakulf — APPROVED** [review]
Fix recived vs recieved vs received.

**Review by flodolo — APPROVED** [review]

**Pull request merged**